### ISSUE TITLE
歌手のみのキャラクターが`characterInfos`の一番後ろに配置されてしまう

### ISSUE THREAD

**Issue by sabonerune** [VOICEVOX/voicevox#2079]
## 不具合の内容

ref: https://github.com/VOICEVOX/voicevox/pull/1973#discussion_r1566392930
現在エンジンからのキャラクター取得は`/speakers`エンドポイントと`/singers`エンドポイントから取得した後それをマージしています。
しかし話者・歌手両方を含めた順序をエンジンから取得する方法が存在しないためエンジンが意図する順序にすることができません。

そもそも話者と歌手情報を一つにまとめていること自体に問題があるかもしれません。

### 再現手順

現状歌手のみのキャラクターが存在しないため再現不可

### 期待動作

`characterInfos`に格納されるキャラクターがエンジンの意図したとおりになる。

## その他

#2078 と関係あるかも。

**Comment by qryxip** [comment]
ENGINEが吐く、というよりCOREのresourceに入れてからENGINEで対応というのはどうでしょうか? 現CORE v0.15はmetas.jsonの中身に一切関与しないので、metas.jsonを差し替えるだけでよいはず。この問題の優先度次第ではありますが。

CC: @tarepan

**Comment by tarepan** [comment]
エディタは [`SpeakerInfo`](https://github.com/VOICEVOX/voicevox_engine/blob/master/voicevox_engine/metas/Metas.py#L81) に `order` 的な属性を持たせて返してほしい、だと理解しました。  

「metas.jsonを差し替える」は「CORE の [metas](https://github.com/VOICEVOX/voicevox_engine/blob/465d3698a326ff61167e34879b2f9656c08f4b84/voicevox_engine/core/core_adapter.py#L26) API が `order` 持ちの json を返すようにする」で理解合っているでしょうか？  
エンジン実装上の問題はとくになさそうです。API 変更ではあるので、色々丁寧に進行する必要はあると思います。

**Comment by qryxip** [comment]
> エディタは [`SpeakerInfo`](https://github.com/VOICEVOX/voicevox_engine/blob/master/voicevox_engine/metas/Metas.py?rgh-link-date=2024-05-14T16%3A01%3A54Z#L81) に `order` 的な属性を持たせて返してほしい、だと理解しました。

そうです。`order: int | None = None`のようになるはずです (VOICEVOX/voicevox_core#728 でのPython APIでは実際にそうしています)。

> 「metas.jsonを差し替える」は「CORE の [metas](https://github.com/VOICEVOX/voicevox_engine/blob/465d3698a326ff61167e34879b2f9656c08f4b84/voicevox_engine/core/core_adapter.py#L26) API が `order` 持ちの json を返すようにする」で理解合っているでしょうか？ 

ENGINEから見たらそうなります。

> API 変更ではあるので、色々丁寧に進行する必要はあると思います。

"metas"の仕様として VOICEVOX/voicevox_core#728 内で議論しているので、それを延長する形になるかと思います。

**Comment by qryxip** [comment]
> 現CORE v0.15はmetas.jsonの中身に一切関与しないので、metas.jsonを差し替えるだけでよいはず。

よく考えたらCOREのビルドは必要ですね。現状のリリース体制だとCOREとモデルは不可分なので新しいバージョンを出す必要があり、前のバージョンのバイナリを使い回そうにもC APIには`voicevox_get_version`が…

まあ次のキャラクター追加/更新のとき(CORE v0.16よりかは早いはず)に設定するのがいいのかも？

**Comment by Hiroshiba** [comment]
全然関係ないんですが、ソングがあるキャラクターを優先的に上に表示すればいいのではと思いましたが、結局ソングのみのキャラクターの後にソング＋トークのあるキャラクターが実装されたら順番が変わっちゃいますね。。

speaker.orderを使うのが丸いなと思いました！！

実装早めるかどうかですが、まあちょっと VVM のタイミングで全部一気にやっちゃいたいというのと（優先度が高いのがたくさんある）、あと問題が出てくるのが「ソングのみのキャラクターがリリースされた後にトークキャラクターをリリースしたとき」で、まあそういうことがあるにしても相当先になるんじゃないんかなと思いました！
ということでCORE v0.16の後にエンジンに実装→エディタに実装が丸いのではないかなと････････！！！
